trin-history/src/events.rs

```diff
@@ -3,35 +3,43 @@ use discv5::TalkRequest;
 use std::sync::Arc;
 use tokio::sync::mpsc::UnboundedReceiver;
 use tracing::{debug, error, warn, Instrument};
-use trin_core::portalnet::types::messages::Message;
+use trin_core::{portalnet::types::messages::Message, utp::stream::UtpListenerEvent};
 
 pub struct HistoryEvents {
     pub network: Arc<HistoryNetwork>,
     pub event_rx: UnboundedReceiver<TalkRequest>,
+    pub utp_listener_rx: UnboundedReceiver<UtpListenerEvent>,
 }
 
 impl HistoryEvents {
-    pub async fn process_requests(mut self) {
-        while let Some(talk_request) = self.event_rx.recv().await {
-            let reply = match self
-                .network
-                .overlay
-                .process_one_request(&talk_request)
-                .instrument(tracing::info_span!("history_network"))
-                .await
-            {
-                Ok(response) => {
-                    debug!("Sending reply: {:?}", response);
-                    Message::from(response).into()
+    pub async fn start(mut self) {
+        loop {
+            tokio::select! {
+                Some(talk_request) = self.event_rx.recv() => {
+                let reply = match self
```

Comment:
*nit* shouldn't this block be indented once more

Reply:
Seems like `cargo fmt` doesn't know how to handle `select!` properly? The other select was funky too.

One way to avoid the issue is by doing:
```rs
Some(talk_request) = self.event_rx.recv() => self.handle_history_talk_request(talk_request),